modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableHeaderRow.java

```diff
@@ -378,9 +399,10 @@ private final void setRootHeader(NestedTableColumnHeader value) {
         return snappedTopInset() + headerPrefHeight + snappedBottomInset();
     }
 
-    // used to be protected to allow subclasses to modify the horizontal scrolling,
-    // but made private again for JDK 9
-    void updateScrollX() {
+       /**
+     * This method is called whenever a scroll is happening on the horizontal bar.
+     */
+    protected void updateScrollX() {
```

Comment:
Each time a scroll is happening, this method is called in order to shift the TableColumnHeaders. 
If anyone has overriden the TableHeader to provide another functionality, it must also be warned that a scroll is happening.

For example, you can add some icons above each TableColumnHeader, or even remove the columnHeader and simply have some icons. But you would want them to slide when the scrollBar is touched, thus having access to updateScrollX().

![pickers](https://user-images.githubusercontent.com/8961645/45475980-20be5500-b73e-11e8-8589-b94a983b0fcf.PNG)